ext/notify/webhook/webhook.go

```diff
@@ -0,0 +1,161 @@
+package webhook
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
+	"github.com/goto/optimus/core/scheduler"
+)
+
+const (
+	httpChannelBuffer         = 100
+	DefaultEventBatchInterval = time.Second * 10
+)
+
+var (
+	notifierType        = "webhook"
+	webhookQueueCounter = promauto.NewCounter(prometheus.CounterOpts{
+		Name:        scheduler.MetricNotificationQueue,
+		ConstLabels: map[string]string{"type": notifierType},
+	})
+
+	webhookWorkerSendErrCounter = promauto.NewCounter(prometheus.CounterOpts{
+		Name:        scheduler.MetricNotificationWorkerSendErr,
+		ConstLabels: map[string]string{"type": notifierType},
+	})
+
+	webhookWorkerSendCounter = promauto.NewCounter(prometheus.CounterOpts{
+		Name:        scheduler.MetricNotificationSend,
+		ConstLabels: map[string]string{"type": notifierType},
+	})
+)
+
+type Notifier struct {
+	io.Closer
+
+	eventChan     chan event
+	wg            sync.WaitGroup
+	workerErrChan chan error
+
+	eventBatchInterval time.Duration
+}
+
+type event struct {
+	url     string
+	headers map[string]string
+	meta    *scheduler.Event
+	jobMeta *scheduler.JobRunMeta
+}
+
+type webhookPayload struct {
+	JobName     string            `json:"job_name"`
+	Project     string            `json:"project"`
+	Namespace   string            `json:"namespace"`
+	Destination string            `json:"destination"`
+	ScheduledAt string            `json:"scheduled_at"`
+	Status      string            `json:"status"`
+	JobLabel    map[string]string `json:"job_label,omitempty"`
+}
+
+func (s *Notifier) Notify(_ context.Context, attr scheduler.WebhookAttrs) error { //nolint:gocritic,unparam
+	go func() {
+		s.eventChan <- event{
+			url:     attr.Route,
+			meta:    attr.JobEvent,
+			jobMeta: attr.Meta,
+			headers: attr.Headers,
+		}
+	}()
+
+	webhookQueueCounter.Inc()
```

Comment:
counter will increment before the event is sent on the channel. This is sync, sending is async.